test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,166 @@
+
+Author(s):
+Norton Luo
+
+'''
+import json
+from time import sleep
+import os
+import sys
+import json
+import time
+import string
+from datetime import *
+import time
+import subprocess
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/common")
+import fit_common
+import test_api_utils
+import fit_amqp
+
+amqp_message_received = False
+routingkey = ""
+amqpbody = ""
+
+
+def get_node_list_by_type(type):
+    mon_url = '/api/2.0/nodes'
+    response = fit_common.rackhdapi(mon_url)
+    nodefound = []
+    if response['status'] in range(200, 205):
+        for nodes in response['json']:
+            if nodes["type"] == type:
+                nodefound.append(nodes["id"])
+    if fit_common.VERBOSITY >= 2:
+        print type," list=", nodefound
+    return nodefound
+
+
+def amqpcallback(ch, method, properties, body):
+    if fit_common.VERBOSITY >= 2:
+        print(" Routing Key %r:" % (method.routing_key))
+        print body
+    global amqp_message_received
+    global routingkey, amqpbody
+    amqp_message_received = True
+    amqpbody = body
+    routingkey = method.routing_key
+
+
+# Select nose.plugins.attrib import attr
+from nose.plugins.attrib import attr
+
+
+@attr(all=True, regression=True, smoke=True)
+class amqp_rack_node(fit_common.unittest.TestCase):
+    # clear the test environment
+    def tear_down(self):
+        self.test_api_delete_rack()
+
+    def compare_message(self, expectedkey, expectedpayload):
+        global routingkey, amqpbody
+        self.assertEquals(routingkey, expectedkey, "No AMQP message received")
+        try:
+            amqp_body_json = fit_common.json.loads(amqpbody)
+        except:
+            print "FAILURE - The message body is not json format! "
+            return False
+        try:
+            self.assertEquals(amqp_body_json['version'], expectedpayload['version'], "version field not correct!")
+            self.assertEquals(amqp_body_json['typeId'], expectedpayload['typeId'], "typeId field not correct!")
+            self.assertEquals(amqp_body_json['action'], expectedpayload['action'], "action field not correct!")
+            self.assertEquals(amqp_body_json['severity'], expectedpayload['severity'], "serverity field not correct!")
+            self.assertNotEquals(amqp_body_json['createdAt'], {}, "createdAt field not correct!")
+            self.assertNotEquals(amqp_body_json['data'], {}, "createdAt field not correct!")
+        except ValueError:
+            print "FAILURE - expected key is missing in the AMQP message!"
+            return False
+        return True
+
+    def test_api_create_and_check_racks(self):
+        for operator in range(0, 10):
+            rackname = "myRack" + "_" + datetime.now().__str__()
+            newrack = {"name": rackname, "type": "rack"}
+            if fit_common.VERBOSITY >= 2:
+                print "create new tag: ", newrack

```

Comment:
So more detailed info on logging: you have a range from .debug_0 to .debug_9. .debug is the same as .debug_2. By default .debug_0 to .debug_5 is captured into the log files. So, if you have a loop like this, you can:
logs.debug_2("Going to generate 10 racks of amqp messages")
for operator in range(0,10):...

Then: logs.debug_3("  Create new rack tag: %s", newrack)

and so on.